### DISCUSSION TITLE
Python 3.9 requirement

### DISCUSSION THREAD

**Discussion by hchoi-moveworks** [ELS-RD/kernl#275] [Q&A]
Is there a specific reason for python 3.9 requirement? We are trying to optimize huggingface `T5ForConditionalGeneration` and was wondering which part of the codebase requires python 3.9.

**Comment by pommedeterresautee** [comment]
Triton :-) 
I think they have a plan to build for 3.10 but not yet merged

**Comment by hchoi-moveworks** [comment]
Thanks @pommedeterresautee  for the response! :) 

From [1], 
>>  Binary wheels are available for CPython 3.6-3.9 and PyPy 3.6-3.7.


Would this mean that `kernl` library is compatible with Cpython 3.6-3.9 as well?  (Not just 3.9)? 

[1]: https://triton-lang.org/master/getting-started/installation.html

**Reply by pommedeterresautee** [reply]
3.6 and 3.7, sure not possible because of PyTorch 2.0 not supporting those I think (not checked and I am on my phone)
3.8 we have not tried

**Reply by pommedeterresautee** [reply]
3.9 is declared in the setup.py, you need to modify that file to pass that check.

https://github.com/ELS-RD/kernl/blob/main/setup.py
```python
    python_requires="==3.9.*",
```
